db/error_handler.cc

```diff
@@ -238,6 +238,33 @@ Status ErrorHandler::SetBGError(const Status& bg_err, BackgroundErrorReason reas
   return bg_error_;
 }
 
+Status ErrorHandler::SetBGError(const IOStatus& bg_io_err,
+                                BackgroundErrorReason reason) {
+  db_mutex_->AssertHeld();
+  if (bg_io_err.ok()) {
+    return Status::OK();
+  }
+  if (recovery_in_prog_ && recovery_error_.ok()) {
+    recovery_error_ = bg_io_err;
+  }
+  Status new_bg_io_err = bg_io_err;
+  Status s;
+  // First, check if the error is a retryable IO error or not.
+  if (bg_io_err.GetRetryable()) {
+    // In current stage, treat retryable error as HardError. No automatic
+    // recovery.
+    bool auto_recovery = false;
+    Status bg_err(new_bg_io_err, Status::Severity::kHardError);
+    bg_error_ = bg_err;
```

Comment:
We might end up overwriting a more severe error with a less severe one. Check the current severity of bg_error_ before overwriting.

Reply:
@anand1976 I see. I missed this possibility. Thanks!